### PR TITLE
ref: collapse middlewares of GET /:formId/adminform/submissions route

### PR DESCRIPTION
**Pull request by karrui**
## Problem

This PR cleans up the janky `authEncryptedResponseAccess` middleware used in the `/adminform/submissions` endpoint and collapses the used middlewares in the `handleGetEncryptedResponse` controller handler.

This PR is just one of the few PRs needed to remove total usage of the `authEncryptedResponseAccess` middleware.

Related to #1434

## Solution

**Improvements**:

- feat: inline auth middlewares for `handleGetEncryptedResponse`

## Tests

- unit tests have been added for the refactored controller handler function

### Manual tests
- [ ] Go to view responses for a storage mode form. 
  - [ ] Click into a single response without attachments. The data should decrypt properly and show no attachments to download.
  - [ ] Click into a single response with attachments. The data should decrypt properly and show attachments to download.

### CODE REVIEW

File: src/app/modules/submission/encrypt-submission/encrypt-submission.controller.ts [hunk 1]
```diff
@@ -524,75 +526,69 @@ export const handleStreamEncryptedResponses: RequestHandler<
...
-    })
-
-    const { statusCode, errorMessage } = mapRouteError(submissionResult.error)
-    return res.status(statusCode).json({
-      message: errorMessage,
-    })
-  }
+  return (
+    // Step 1: Retrieve logged in user.
+    getPopulatedUserById(sessionUserId)
```
Comment (seaerchin) [inline src/app/modules/submission/encrypt-submission/encrypt-submission.controller.ts:539]: think i need context here; why is this checked now when it wasn't previously ah

**Review by seaerchin — COMMENTED** [review]
looks good!

File: src/app/modules/submission/encrypt-submission/encrypt-submission.controller.ts [hunk 2]
```diff
@@ -524,75 +526,69 @@ export const handleStreamEncryptedResponses: RequestHandler<
...
+            refNo: submissionData._id,
+            submissionTime: moment(submissionData.created)
+              .tz('Asia/Singapore')
+              .format('ddd, D MMM YYYY, hh:mm:ss A'),
+            content: submissionData.encryptedContent,
+            verified: submissionData.verifiedContent,
+            attachmentMetadata: presignedUrls,
+          }
+          return responseData
+        })
```
Comment (seaerchin) [inline src/app/modules/submission/encrypt-submission/encrypt-submission.controller.ts:572]: nit: maybe can just make this a utility method 
```ts

const generateEncryptedSubmissionDto = (submissionData, presignedUrls) => {blah}
```

main benefit would be readability

File: src/app/modules/submission/encrypt-submission/__tests__/encrypt-submission.controller.spec.ts [hunk 3]
```diff
@@ -88,10 +115,114 @@ describe('encrypt-submission.controller', () => {
...
+    it('should return 400 if form is not an encrypt mode form', async () => {
+      // Arrange
+      const mockRes = expressHandler.mockResponse()
+
+      MockUserService.getPopulatedUserById.mockReturnValueOnce(
+        okAsync(MOCK_USER),
+      )
+      MockAuthService.getFormAfterPermissionChecks.mockReturnValueOnce(
+        okAsync(MOCK_FORM),
+      )
```
Comment (mantariksh) [inline src/app/modules/submission/encrypt-submission/__tests__/encrypt-submission.controller.spec.ts:142]: again, might be good to put this in a `beforeEach` for readability

File: src/app/modules/submission/encrypt-submission/encrypt-submission.controller.ts [hunk 4]
```diff
@@ -524,75 +526,69 @@ export const handleStreamEncryptedResponses: RequestHandler<
...
+      })
+      .map((responseData) => res.json(responseData))
+      .mapErr((error) => {
+        logger.error({
+          message: 'Failure retrieving encrypted submission response',
+          meta: {
+            action: 'handleGetEncryptedResponse',
+            submissionId,
+            formId,
```
Comment (mantariksh) [inline src/app/modules/submission/encrypt-submission/encrypt-submission.controller.ts:581]:
```suggestion
            formId,
            ...createReqMeta(req),
```

File: src/app/modules/submission/encrypt-submission/encrypt-submission.controller.ts (hunk 1 above)
Comment (mantariksh) [reply src/app/modules/submission/encrypt-submission/encrypt-submission.controller.ts:539]: if you look at the changes in `admin-form.routes`, there used to be a function called `authEncryptedResponseAccess` which did this

**Review by mantariksh — APPROVED** [review]

File: src/app/modules/submission/encrypt-submission/encrypt-submission.controller.ts (hunk 1 above)
Comment (mantariksh) [reply src/app/modules/submission/encrypt-submission/encrypt-submission.controller.ts:539]: if you look at the changes in `admin-form.routes`, there used to be a function called `authEncryptedResponseAccess` which did this

File: src/app/modules/submission/encrypt-submission/encrypt-submission.controller.ts (hunk 4 above)
Comment (karrui) [reply src/app/modules/submission/encrypt-submission/encrypt-submission.controller.ts:581]: wonderful, thanks for the catch

File: src/app/modules/submission/encrypt-submission/__tests__/encrypt-submission.controller.spec.ts (hunk 3 above)
Comment (karrui) [reply src/app/modules/submission/encrypt-submission/__tests__/encrypt-submission.controller.spec.ts]: Resolved in 2e97f93

**Pull request merged**